### PR TITLE
Refresh the notification panel appearance

### PR DESCRIPTION
**Pull request by SimonBrandner**
Type: enhancement

<hr>

Screenshot_20210824_124933

---
Here's what your changelog entry will look like:

## ✨ Features
 * Refresh the notification panel appearance ([\#6668](https://github.com/matrix-org/matrix-react-sdk/pull/6668)).

Preview: https://pr6668--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/rooms/EventTile.tsx [hunk 1]
```diff
@@ -1075,18 +1086,17 @@ export default class EventTile extends React.Component<IProps, IState> {
...
+                    <div
+                        className="mx_EventTile_roomName"
+                        key="mx_EventTile_roomName"
+                        onClick={this.onRoomClicked}
+                    >
                         <RoomAvatar room={room} width={28} height={28} />
-                        <a href={permalink} onClick={this.onPermalinkClicked}>
-                            { room ? room.name : '' }
-                        </a>
+                        { room ? room.name : '' }
```
Comment (turt2live) [inline src/components/views/rooms/EventTile.tsx:1095]: should be able to change this to `room?.name`

**Review by turt2live — APPROVED** [review]

File: src/components/views/rooms/EventTile.tsx [hunk 2]
```diff
@@ -1092,7 +1092,7 @@ export default class EventTile extends React.Component<IProps, IState> {
                         onClick={this.onRoomClicked}
                     >
                         <RoomAvatar room={room} width={28} height={28} />
-                        { room ? room.name : '' }
+                        { room?.name || '' }
```
Comment (turt2live) [inline src/components/views/rooms/EventTile.tsx:1095]: you also shouldn't need the `||` anymore, because undefined/null gets rendered as an empty string.

Comment (SimonBrandner) [reply src/components/views/rooms/EventTile.tsx]: :facepalm:

**Comment by ShadowJonathan** [comment]
I'd recommend;

- Having "view message" only be visible when hovering over the message
- Try to fetch the username in that room and display it (though maybe with the mxid label next to it)
- Have the room avatar be smaller (almost as high as the room name), while showing the user avatar as small or bigger next to the username

**Comment by SimonBrandner** [comment]
> Try to fetch the username in that room and display it (though maybe with the mxid label next to it)

That is a separate issue which I don't know how to solve just yet

**Comment by williamkray** [comment]
i think for readability, showing the avatar and display nick of the person who said the thing first, followed by `in (Room Name)` where `(Room Name)` is perhaps a pill.

**Review by janogarcia — COMMENTED** [review]
@SimonBrandner The notification panel is in need of a design overhaul. The current one is just terrible. I’m afraid the changes you’re introducing are not necessarily fixing the core issues or making it strictly better. (e.g. more visual noise, tighter/crammed spacing, legacy panel header...).

As a first step towards fixing that massive debt, I’d suggest trying to implement the incremental improvements I explored in the Threads project a while ago:

[Threads — Notifications](https://www.figma.com/file/T309ztx0sNyOOK6NKVLHsK/Threads?node-id=2716%3A265132&t=RAry6Jkdg93yQvMI-0)

It's not fixing all the issues, but it would be at least a first solid step.

After implementing those improvements, we could focus on all the remaining issues:

- Display names and avatar for users
- Grouping notifications by room
- Ordering chronologically, from most recent, top to bottom
- etc.

---

Frame 6492 (1)

**Comment by SimonBrandner** [comment]
Since the PR seems far away from what we want and I lack the time to implement what we want atm, I am going to close it, at least for now

**Pull request closed without merging**